### PR TITLE
Remove error string comparisons from tests

### PR DESCRIPTION
**Pull request by mchompalova**

### CODE REVIEW

File: concept/read_api_test.go [hunk 1]
```diff
@@ -119,7 +122,7 @@ func TestGetConceptsByIDsNon200HTTPStatus(t *testing.T) {
 
 	ctx := tidUtils.TransactionAwareContext(context.Background(), tidUtils.NewTransactionID())
 	_, err := csAPI.GetConceptsByIDs(ctx, []string{"an-id"})
-	assert.EqualError(t, err, "concept search API returned a non-200 HTTP status code: 503")
+	assert.Error(t, err)
```
Comment (ivan-p-nikolov) [inline concept/read_api_test.go]: We could define an error in read_api.go and wrap it when returning. Something like:
```go
var ErrUnexpectedResponse = errors.New("concept search API returned a non-200 HTTP status code")
//....
if resp.StatusCode != http.StatusOK {
	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		err = fmt.Errorf("Status %d : %w", resp.Status, ErrUnexpectedResponse)
	} else {
		err = fmt.Errorf("Status %d  Response %s : %w", resp.Status, string(body), ErrUnexpectedResponse)
	}
	batchConceptsLog.WithError(err).Error("Error received from concept search API")
	return nil, err
}
```

File: handler/handler_test.go [hunk 2]
```diff
@@ -688,8 +688,7 @@ func TestFetchFromAnnotationsAPIWithInvalidURL(t *testing.T) {
 
 	assert.Equal(t, http.StatusInternalServerError, resp.StatusCode)
 	assert.NoError(t, err)
-	assert.JSONEq(t, "{\"message\":\"Failed to read annotations: parse :: missing protocol scheme\"}", string(body))
-
+	assert.Contains(t, string(body), "Failed to read annotations")
```
Comment (ivan-p-nikolov) [inline handler/handler_test.go]: Here as well. We can define an error and wrap it. That way we wouldn't degrade the test.

File: annotations/annotations_rw_test.go [hunk 3]
```diff
@@ -90,7 +91,8 @@ func TestReadHTTPRequestError(t *testing.T) {
 	rw := NewRW(testClient, ":#")
 	ctx := tidUtils.TransactionAwareContext(context.Background(), tid)
 	_, _, found, err := rw.Read(ctx, testContentUUID)
-	assert.EqualError(t, err, "parse :: missing protocol scheme")
+	assert.Error(t, err.(*url.Error))
```
[the remark this answers is not in the document]
Comment (mchompalova) [reply annotations/annotations_rw_test.go]: Actually, errors.As is what we need in this case :)

File: handler/handler_test.go (hunk 2 above)
Comment (mchompalova) [reply handler/handler_test.go]: Here, we compare strings hence I can't use errors.Is to check for the error type.

File: annotations/annotations_rw.go [hunk 4]
```diff
@@ -72,7 +79,7 @@ func (rw *annotationsRW) Read(ctx context.Context, contentUUID string) (*Annotat
 	case http.StatusNotFound:
 		return nil, "", false, nil
 	default:
-		return nil, "", false, fmt.Errorf("annotations RW returned an unexpected HTTP status code in read operation: %v", resp.StatusCode)
+		return nil, "", false, fmt.Errorf("status %v: %w", resp.StatusCode, ErrUnexpectedStatusRead)
```
Comment (ivan-p-nikolov) [inline annotations/annotations_rw.go]: Let use `%d` for the status code.

File: annotations/annotations_rw.go [hunk 5]
```diff
@@ -128,19 +135,24 @@ func (rw *annotationsRW) GTG() error {
 	req, err := http.NewRequest("GET", rw.endpoint+"/__gtg", nil)
 	if err != nil {
 		log.WithError(err).Error("Error in creating the HTTP request to annotations RW GTG")
-		return fmt.Errorf("gtg HTTP request error: %v", err)
+		return fmt.Errorf("%w %v", ErrGTGRequest, err)
```
Comment (ivan-p-nikolov) [inline annotations/annotations_rw.go]: It's standard in golang when wrapping an error to be at the end of the string and be separate by `:` from the rest of the error message. So `"%v : %w"`.

File: concept/read_api_test.go [hunk 6]
```diff
@@ -133,7 +140,7 @@ func TestGetConceptsByIDsUnmarshallingPayloadError(t *testing.T) {
 
 	ctx := tidUtils.TransactionAwareContext(context.Background(), tidUtils.NewTransactionID())
 	_, err := csAPI.GetConceptsByIDs(ctx, []string{"an-id"})
-	assert.EqualError(t, err, "invalid character '}' looking for beginning of value")
+	assert.Error(t, err)
```
Comment (ivan-p-nikolov) [inline concept/read_api_test.go]: I suspect that the returned error is some form of json error. Something like `json.SintaxError` We can check the documentations. 
So we can test with `errors.As` for it.

Comment (karsov) [inline concept/read_api_test.go]: I think we can check for https://golang.org/pkg/encoding/json/#InvalidUnmarshalError
... or https://golang.org/pkg/encoding/json/#SyntaxError - I saw you use the last one in another place.

File: concept/read_api_test.go [hunk 7]
```diff
@@ -177,7 +184,7 @@ func TestUnhappyGTG(t *testing.T) {
 	csAPI := NewReadAPI(testClient, s.URL, apiKey, batchSize)
 
 	err := csAPI.GTG()
-	assert.EqualError(t, err, "concept search API returned a non-200 HTTP status code: 503")
+	assert.Error(t, err)
```
Comment (karsov) [inline concept/read_api_test.go]: We should be able to use "assert.True(t, errors.Is(err, ErrUnexpectedResponse))"

File: handler/handler_test.go (hunk 2 above)
Comment (karsov) [reply handler/handler_test.go]: This is a URL parsing error we can check for *url.Error

File: annotations/annotations_rw.go (hunk 4 above)
Comment (mchompalova) [reply annotations/annotations_rw.go]: In the original code, it was %v, that's why I tried to preserve it.

File: annotations/annotations_api_test.go [hunk 8]
```diff
@@ -34,7 +34,7 @@ func TestUnhappyAnnotationsAPIGTG(t *testing.T) {
 
 	annotationsAPI := NewUPPAnnotationsAPI(testClient, annotationsServerMock.URL+"/content/%v/annotations", testAPIKey)
 	err := annotationsAPI.GTG()
-	assert.EqualError(t, err, "gtg returned a non-200 HTTP status [503]: I am not happy!")
+	assert.Error(t, err)
```
Comment (karsov) [inline annotations/annotations_api_test.go]: This is not a text that depends on an external library, so we can check for it.

File: annotations/annotations_api_test.go [hunk 9]
```diff
@@ -43,13 +43,13 @@ func TestAnnotationsAPIGTGWrongAPIKey(t *testing.T) {
 
 	annotationsAPI := NewUPPAnnotationsAPI(testClient, annotationsServerMock.URL+"/content/%v/annotations", "a-non-existing-key")
 	err := annotationsAPI.GTG()
-	assert.EqualError(t, err, "gtg returned a non-200 HTTP status [401]: unauthorized")
+	assert.Error(t, err)
```
Comment (karsov) [inline annotations/annotations_api_test.go]: This is not a text that depends on an external library, so we can check for it.

File: annotations/annotations_api_test.go [hunk 10]
```diff
@@ -43,13 +43,13 @@ func TestAnnotationsAPIGTGWrongAPIKey(t *testing.T) {
...
 }
 
 func TestAnnotationsAPIGTGInvalidURL(t *testing.T) {
 	annotationsAPI := NewUPPAnnotationsAPI(testClient, ":#", testAPIKey)
 	err := annotationsAPI.GTG()
-	assert.EqualError(t, err, "gtg request error: parse :: missing protocol scheme")
+	assert.Error(t, err)
```
Comment (karsov) [inline annotations/annotations_api_test.go]: I think we can check for url.Error.

File: annotations/annotations_rw_test.go [hunk 11]
```diff
@@ -264,21 +276,21 @@ func TestRWHappyGTG(t *testing.T) {
 func TestRWHTTPRequestErrorGTG(t *testing.T) {
 	rw := NewRW(testClient, ":#")
 	err := rw.GTG()
-	assert.EqualError(t, err, "gtg HTTP request error: parse :: missing protocol scheme")
+	assert.True(t, errors.Is(err, ErrGTGRequest))
```
Comment (karsov) [inline annotations/annotations_rw_test.go]: To be precise we can check for url.Error too.

File: annotations/annotations_rw_test.go [hunk 12]
```diff
@@ -264,21 +276,21 @@ func TestRWHappyGTG(t *testing.T) {
...
 }
 
 func TestRWHTTPCallErrorGTG(t *testing.T) {
 	rw := NewRW(testClient, "")
 	err := rw.GTG()
-	assert.EqualError(t, err, "gtg HTTP call error: Get /__gtg: unsupported protocol scheme \"\"")
+	assert.True(t, errors.Is(err, ErrGTGCall))
```
Comment (karsov) [inline annotations/annotations_rw_test.go]: To be precise we can check for url.Error too.

File: concept/read_api_test.go (hunk 6 above)
Comment (mchompalova) [reply concept/read_api_test.go]: I will fix it.

File: annotations/annotations_rw_test.go [hunk 13]
```diff
@@ -264,21 +281,27 @@ func TestRWHappyGTG(t *testing.T) {
...
+	assert.True(t, errors.As(e, &urlError))
 }
 
 func TestRWHTTPCallErrorGTG(t *testing.T) {
 	rw := NewRW(testClient, "")
 	err := rw.GTG()
-	assert.EqualError(t, err, "gtg HTTP call error: Get /__gtg: unsupported protocol scheme \"\"")
+	e := errors.Unwrap(err)
+	var urlError *url.Error
+	assert.True(t, errors.As(err, &ErrGTGCall))
```
Comment (ivan-p-nikolov) [inline annotations/annotations_rw_test.go:295]: Here you change the value of `ErrGTGCall` and test nothing. I would suggest to wrap `ErrGTGCall` and not `url.Error` and test with `errors.Is`

File: annotations/annotations_rw_test.go [hunk 14]
```diff
@@ -264,21 +281,27 @@ func TestRWHappyGTG(t *testing.T) {
 func TestRWHTTPRequestErrorGTG(t *testing.T) {
 	rw := NewRW(testClient, ":#")
 	err := rw.GTG()
-	assert.EqualError(t, err, "gtg HTTP request error: parse :: missing protocol scheme")
+	e := errors.Unwrap(err)
+	var urlError *url.Error
+	assert.True(t, errors.As(err, &ErrGTGRequest))
```
Comment (ivan-p-nikolov) [inline annotations/annotations_rw_test.go:286]: Same as `TestRWHTTPCallErrorGTG`

File: annotations/annotations_api_test.go [hunk 15]
```diff
@@ -43,13 +45,16 @@ func TestAnnotationsAPIGTGWrongAPIKey(t *testing.T) {
...
+	assert.True(t, errors.Is(err, ErrGTGNotOK))
 }
 
 func TestAnnotationsAPIGTGInvalidURL(t *testing.T) {
 	annotationsAPI := NewUPPAnnotationsAPI(testClient, ":#", testAPIKey)
 	err := annotationsAPI.GTG()
-	assert.EqualError(t, err, "gtg request error: parse :: missing protocol scheme")
+	e := errors.Unwrap(err)
+	var urlErr *url.Error
+	assert.True(t, errors.As(err, &ErrGTGRequest))
```
Comment (ivan-p-nikolov) [inline annotations/annotations_api_test.go:56]: Same as `TestRWHTTPCallErrorGTG`

File: annotations/annotations_rw_test.go [hunk 16]
```diff
@@ -112,7 +121,9 @@ func TestReadInvalidBodyError(t *testing.T) {
 	rw := NewRW(testClient, s.URL)
 	ctx := tidUtils.TransactionAwareContext(context.Background(), tid)
 	_, _, found, err := rw.Read(ctx, testContentUUID)
-	assert.EqualError(t, err, "invalid character 'i' looking for beginning of object key string")
+	var jsonErr *json.SyntaxError
+	assert.True(t, errors.As(err, &jsonErr))
```
Comment (ivan-p-nikolov) [inline annotations/annotations_rw_test.go:125]: `As` will panic if target is nil or not a pointer. Test if the `err` is `nil` first and then use `As`. 
```suggestion
	if (assert.Error(t, err)) {
		assert.True(t, errors.As(err, &jsonErr))
	}
```

**Pull request merged**